Update Deno Deploy guide (#1081)
grammyjs/website: site/docs/hosting/deno-deploy.md

   > It is recommended that you have a single stable branch and you do your testing stuff in other branches, so that you won't get some unexpected things happen.

3. Visit your [Deno Deploy dashboard](https://dash.deno.com/account/overview).
4. Click on "New Project".
5. Install the GitHub app on your account or organization, and choose your repository.
6. Select the branch you want to deploy.
7. Select the entrypoint file `main.ts`, and click "Deploy Project" to deploy.

### Method 2: With `deployctl`

> This is a method for more advanced users or if you don't want to upload your code to GitHub.
> It allows you to deploy the project via the command line or Github Actions.

1. Install [`deployctl`](https://github.com/denoland/deployctl).

> 2. Create an access token from the "Access Tokens" section in [account settings](https://dash.deno.com/account).

Author: The `access-token` anchor was removed due to the page's lazy loading behavior, so that link-checker doesn't have to be configured to check for the anchor. I think the information is good enough. But if it needs to be reverted, let me know, I can adjust link-checker to ignore this anchor.

3. Go to your project directory and run the following command:
   ```sh:no-line-numbers
   deployctl deploy --project=<project> --entrypoint=./main.ts --prod --token=<token>
   ```

   ::: tip Setting environment variables
   Environment variables can be set by heading to your project's settings after deploying.

>    But this is also possible from the command line as well:

Reviewer: ```suggestion
   But this is possible from the command line, as well:
```

   1. You can assign environment variables from a dotenv file by adding the `--env-file=<file>` argument.
   2. You can also specify them individually by using the `--env=<key=value>` argument.

   :::

> 4. To set up Github Actions, refer to [this](https://github.com/denoland/deployctl/blob/main/action/README.md).

Reviewer: ```suggestion
4. To set up GitHub Actions, refer to [this](https://github.com/denoland/deployctl/blob/main/action/README.md).
```

> Checkout the [deployctl documentation](https://docs.deno.com/deploy/manual/deployctl) for more information.

Reviewer: ```suggestion
Check out the [deployctl documentation](https://docs.deno.com/deploy/manual/deployctl) for more information.
```

### Note

After getting your app running, you should configure your bot's webhook settings to point to your app.
To do that, send a request to

```http:no-line-numbers

> GET https://api.telegram.org/bot<token>/setWebhook?url=<url>

Reviewer: What do you think of making this a curl command instead?

Author: Hmm, in a way it makes sense, and probably handy for beginners if they are wondering how to make this request.

But thinking the opposite, wouldn't it sound like we are suggesting that they _should_ run the command? I thought this part implied that they should just make this request, by using any method. I usually do this right in the browser.

I'm up for both, and probably overthinking. What do you think, should we change it?

Reviewer: No, let's leave it. Then we can discuss in https://t.me/grammyjs if we should use HTTP notation or a curl command, and follow up with a separate PQ that perform this change consistently across all hosting guides.

Reviewer: https://t.me/grammyjs/248240

Author: Should I change to curl then? According to the votes?

Reviewer: Yep sounds good

Author: Done; sorry it took me a while to get back to this.
